lib/RTCSession.js

```diff
@@ -1046,13 +1047,36 @@ module.exports = class RTCSession extends EventEmitter
       }
       else
       {
-        const dtmf = new RTCSession_DTMF(this);
+        if (transportType !== JsSIP_C.DTMF_TRANSPORT.INFO ||
+          transportType !== JsSIP_C.DTMF_TRANSPORT.IN_BAND)
+        {
+          throw new TypeError(`invalid transportType: ${transportType}`);
+        }
+        // Send DTMF according to transport config.
+        switch (transportType)
+        {
+          case JsSIP_C.DTMF_TRANSPORT.IN_BAND: {
+            // Send DTMF in current audio RTP stream.
+            const sender = this._getDTMFRTPSender();
+
+            if (sender)
+            {
+              sender.insertDTMF(tone, duration, interToneGap);
+            }
 
-        options.eventHandlers = {
-          onFailed : () => { this._tones = null; }
-        };
-        dtmf.send(tone, options);
-        timeout = duration + interToneGap;
+            break;
+          }
+          case JsSIP_C.DTMF_TRANSPORT.INFO: {
```

Comment:
Blank line between `case`. BTW: does hit pass `gulp lint` or not? You did not answer before.

```suggestion
          }

          case JsSIP_C.DTMF_TRANSPORT.INFO: {
```

Reply:
yes it does pass. I will add the blank line as requested